# Fibroblast Signaling Network Model Fitting  
## 10.29.2019 - Jesse Rogers  

This program is intended to estimate reaction parameters for logic-based ODE networks. Using both a network generated by Netflux (https://github.com/saucermanlab/Netflux) and previously-published proteomics and tramscriptomics data for model inputs and outputs respectively, the program iteratively estimates reaction weights using a genetic algorithm (GA). This program can be run either locally or using high-performance computing resources, and is currently adapted to fit a fibroblast signaling network model (SNM)

### Required Software:  
* Matlab 2018
  * Global Optimization Toolbox
  * Parallel Computing Toolbox

### Required Files:  
* Proteomics (model input) data: arranged using Entrez gene symbols as rows, samples as columns, data as log2RFU (for example, see PMID 31511425 SOMAscan proteomics data, Supplementary Table S2)
* Transcriptomics (model output) data: arranged using gene symbols as rows, samples as columns, data as CPM (for example, see PMID RNA-seq data, GSE133529)
* Netflux model: prepared according to Netflux documentation (for example, see snm_1_1_rev3.xlsx)
* PARAMS.txt file: user-editable file containing all needed parameters for running program  

#### Notes:
* Fibroblast SNM: version 1.1, revision 3
  * Based on previously-published SNM (PMID 27017945)
  * Reduced version used for fitting: nodes removed if network logic is unaffected
* Current optimization scheme:
  * ODE network is simulated for 16 sets of input conditions
  * Steady-state output levels are measured and compared to experimentally-measured levels
  * Error is calculated as the mean-square error across all inputs/outputs